lms/static/scripts/frontend_apps/components/GradingControls.tsx

```diff
@@ -38,10 +39,35 @@ export default function GradingControls({
     api: { authToken, sync: syncAPICallInfo },
   } = useConfig();
 
+  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
   const clientRPC = useService(ClientRPC);
   const [selectedStudent, setSelectedStudent] = useState<StudentInfo | null>(
     null
   );
+  const changeSelectedStudent = useCallback(
+    async (newSelectedStudent: StudentInfo | null) => {
+      const continueEditing =
+        hasUnsavedChanges &&
+        (await confirm({
+          title: 'Continue editing?',
+          message: (
+            <div>
+              There are unsaved changes to{' '}
+              <span className="font-bold">{selectedStudent?.displayName}</span>{' '}
+              grade. Do you want to continue editing?
```

Comment:
I will mention the feedback/comment here once this is merged and I have rebased on https://github.com/hypothesis/lms/pull/5643